CMakeLists.txt

```diff
@@ -325,16 +325,43 @@ endif()
 
 option(WITH_BLUEFS "libbluefs library" OFF)
 
-option(WITH_QAT "Enable Qat driver" OFF)
-if(WITH_QAT)
+option(WITH_QATLIB "Enable QAT with qatlib" ON)
```

Comment:
@cbodley pushed to CI to try a Shaman build:
https://shaman.ceph.com/builds/ceph/wip-mk-up-QAT-i001/
there are QAT build errors on _Arm arch_ and _Windows  distro_ since default of `WITH_QATLIB=ON`
but myb there should be an exception for those and set them to `WITH_QATLIB=OFF` by default?

Reply:
> the arm build is failing to build qatlib samples with `error: impossible constraint in 'asm'` - i'll try adding `--disable-samples` to the configure line

[arm64 build](https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=arm64,AVAILABLE_ARCH=arm64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/74425//consoleFull) still failed on:
```
quickassist/lookaside/access_layer/src/qat_direct/common/adf_user_ring.c: In function ‘adf_user_put_msg’:
./quickassist/lookaside/access_layer/src/qat_direct/common/include/adf_platform_common.h:231:9: error: unknown register name ‘%edi’ in ‘asm’
  231 |         __asm__ __volatile__("mov %0, %%esi \n"                                \
      |         ^~~~~~~
```